File: back/Atividade.py
class Atividade:
    def __init__(self, db):
        self.db = db

    def get_atividades(self):
        cursor = self.db.conn.cursor(dictionary=True)
        try:
            cursor.execute("SELECT ID, Descricao FROM atividade")
            result = cursor.fetchall()
            atividades_list = [
                {"ID": atividade[0], "Descricao": atividade[1]} for atividade in result
            ]
            cursor.close()
            return atividades_list
        except Exception as e:
            print(f"Erro ao buscar atividades: {e}")
            cursor.close()
            raise e

File: backjs/classes/Atividade.js
class Atividade {
    constructor(db) {
        this.db = db;
    }

    async getAtividades() {
        try {
            const query = 'SELECT ID, Descricao FROM atividade';
            const result = await this.db.fetchAll(query);
            const atividadesList = result.map(atividade => ({
                ID: atividade.ID,
                Descricao: atividade.Descricao
            }));
            return atividadesList;
        } catch (err) {
            console.error('Erro ao buscar atividades:', err);
            throw err;
        }
    }
}

module.exports = Atividade;